packages/paste-core/components/minimizable-dialog/src/index.tsx

```diff
@@ -1,5 +1,17 @@
-export * from "./MinimizableDialog";
-export * from "./MinimizableDialogButton";
-export * from "./MinimizableDialogHeader";
-export * from "./MinimizableDialogContainer";
-export * from "./MinimizableDialogContent";
+import type { MinimizableDialogStateReturn, UseMinimizableDialogStateArgs } from "./MinimizableDialogContainer";
+
+// for the sake of documenting the types we rename the state hook types to append Props to the name, so we can docuemnt them and not cause a breaking change.
+type MinimizableDialogStateReturnProps = MinimizableDialogStateReturn;
+type UseMinimizableDialogStateArgsProps = UseMinimizableDialogStateArgs;
```

Comment:
suggestion: could use a little alignment here. renaming `MinimizableDialogStateReturnProps` is the return of `useMinimizableDialogue`, but isn't named based on the hook, like the initial state it.

Might be better to use `useMinimizableDialogInitialState` and `useMinimizableDialogReturnState`